ClamAV configuration options

sevein opened the pull request:
This pull request brings some work that @ross-spencer has done in vanilla Archivematica in regards to the antivirus scanning feature (see the [feature file](https://github.com/artefactual-labs/archivematica-acceptance-tests/pull/48) for more details). It solves the problems mentioned in https://jiscdev.atlassian.net/browse/RDSSARK-287.

#### Environment variables before this pull request

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_SERVER`

Clamdscanner only. UNIX socket (if the value starts with `/`) or TCP socket (form `host:port`) where the clamd client should connect to scan files. The default is `/var/run/clamav/clamd.ctl`. A valid value could be `myclamad.dev:3310`. 

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_PASS_BY_REFERENCE`

Clamdscanner only. Whether the file is streamed or referenced. If clamd doesn't have access to the file it should be streamed. The default is `False` (file contents are streamed). 

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_TIMEOUT`

Clamdscanner only. Time before the anti-virus scan is aborted. The default is `600` (seconds). Use integers or floats. 

#### Environment variables added in this pull request

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_CLIENT_BACKEND`

The ClamAV backend used for anti-virus scanning. The two options that are available are: `clamscanner` (via CLI) and `clamdscanner` (over TCP or UNIX socket). The default is `clamdscanner`.

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_MAX_FILE_SIZE`

Files larger than this limit won't be scanned. The default is `42` (MB).

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_MAX_SCAN_SIZE`

This option sets the maximum amount of data to be scanned for each input file. Files larger than this value will be scanned but only up to this limit. Archives and other containers are recursively extracted and scanned up to this value. The default is `42` (MB). Use integers or floats.

sevein committed:
MCPClient: use both clamscan and clamdscan

Added new `CLAMAV_CLIENT_THRESHOLD` setting to MCPClient in order to let the
user define a threshold in MB so when the size of a file being scanned for
viruses is too big the scanner can fall back to clamscan because clamd can't
handle big files not even when the file is passed by reference.

This closes #753.

sevein committed:
MCPClient: install clamav in Docker image

ross-spencer committed:
getfilesize return now gets compared with None

ross-spencer committed:
Implements Antivirus (AV) Configuration Options

* Separates the responsibilities of AV unit tests
* Provides user options for which AV scanner to use
* Provides user options for max_file_size and max_scan_size
* Ensures PREMIS events are not created for unscanned, and ostensibly unscanned files
* Fixes #779
* Resolves #819
* Additional code review changes

sevein committed:
Dockerfile: add defaults to build args

If missing the settings module won't be able to load. I think we should have
defaults because we don't always know exactly what version we should use, e.g.
when we're running the unit tests.

ross-spencer commented:
@sevein (shortened some parts by just saying up-front clamdscanner only. One question below about 86400 second limit - one day?) 

---

#### Environment variables before this pull request

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_SERVER`

Clamdscanner only. UNIX socket (if the value starts with `/`) or TCP socket (form `host:port`) where the clamd client should connect to scan files. The default is `/var/run/clamav/clamd.ctl`. A valid value could be `myclamad.dev:3310`. 

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_PASS_BY_REFERENCE`

Clamdscanner only. Whether the file is streamed or referenced. If clamd doesn't have access to the file it should be streamed. The default is `False` (file contents are streamed). 

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_TIMEOUT`

Clamdscanner only. Time before the anti-virus scan is aborted. The default is `86400` (seconds). Use integers or floats. 

#### Environment variables added in this pull request

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_CLIENT_BACKEND`

The ClamAV backend used for anti-virus scanning. The two options that are available are: `clamscanner` (via CLI) and `clamdscanner` (over TCP or UNIX socket). The default is `clamdscanner`.

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_MAX_FILE_SIZE`

Files larger than this limit won't be scanned. The default is `42` (MB).

`ARCHIVEMATICA_MCPCLIENT_MCPCLIENT_CLAMAV_MAX_SCAN_SIZE`

This option sets the maximum amount of data to be scanned for each input file. Files larger than this value will be scanned but only up to this limit. Archives and other containers are recursively extracted and scanned up to this value. The default is `42` (MB). Use integers or floats.

sevein commented:
> One question below about 86400 second limit - one day?).

Yes that's probably not a great default. I can't remember how I came up with it! What would you suggest?

ross-spencer commented:
It seems to be a familiar number, https://www.google.com/search?q=86400+timeout&oq=86400+timeout&aqs=chrome..69i57j0l4.6356j0j4&sourceid=chrome&ie=UTF-8 for our purposes, do we want to say maybe, 10 minutes? (I wonder if when we get the opportunity we should write a quick unit test as well - what do you think?)

sevein committed:
MCPClient: change ClamAV client timeout to 600s

sevein commented:
Ok done, added new default and updated the description of the PR with your suggestions. Thank you so much!

We're going to have different defaults in JiscRDSS and AM.

sevein commented:
Ready for CR!

sevein commented:
The commits will be squashed, fell free to ignore them, they're arranged like that because the work happened in vanilla over time across a few pull requests.

64b80a8 is addressing a separate concern but it was blocking me so I've added it here to avoid the extra overhead of submitting a second PR.

jhsimpson reviewed (approved):
I haven't tested but we agreed to test this in qa, as it is working in vanilla
